pkg/expr/convert_to_full_long.go

```diff
@@ -160,6 +157,32 @@ func convertTimeSeriesMultiToFullLong(frames data.Frames) (data.Frames, error) {
 		if timeField == nil {
 			return nil, fmt.Errorf("missing time field")
 		}
+	}
+
+	// Calculate total row count to pre-allocate
+	var totalRows int
+	for _, frame := range frames {
+		// Directly count numeric fields since we already validated time fields
+		for _, f := range frame.Fields {
```

Comment:
All rows must be the same length if the input is not malformed, so we can get the length from the frame method.

Reply:
Ok got you, thanks!